### ISSUE TITLE
File picker should have an option to display ignored files

### ISSUE THREAD

**Issue by valignatev** [helix-editor/helix#280]
Helix can't open files that are in the home directory and in some other directories.
Can't quite tell the pattern, but in some directories helix just doesn't open any existing files. As in it doesn't show suggestions. It opens the file if I type its name manually with the relative path, but it can't open the file if I type its path like `~/.vimrc`, e.g. with leading tilde. It says "NOR" and a path to the file
![image](https://user-images.githubusercontent.com/7038954/122245221-36a06c00-cece-11eb-9da3-1df0e3d720d7.png)

### Environment

- Platform: Arch Linux
- Helix version: 0.2.0

Log file is empty

**Comment by pickfire** [comment]
Seemed to work for me, is your `~/.vimrc` empty? I am using arch as well, I tried both v0.2.0 and a364d6c3837c36225dcd4ec9b15ef2c192feef0b. Did you use the one in AUR? How did you install or test it out?

**Comment by valignatev** [comment]
yeah just helix-bin from AUR. My vimrc has content. Again, it opens when I manually type it out like `:open .vimrc`, and shows the content, but it doesn't open it if I type `:open ~/.vimrc`.

**Comment by sudormrfbin** [comment]
> yeah just helix-bin from AUR. My vimrc has content. Again, it opens when I manually type it out like `:open .vimrc`, and shows the content, but it doesn't open it if I type `:open ~/.vimrc`.

I can reproduce this on master. Seems like an issue with expanding `~`.

**Comment by valignatev** [comment]
I believe this isn't fixed still. While ~ gets expanded to the home directory now, no suggestions for the files are shown. 
hx --version is 0.3

**Comment by archseer** [comment]
You need to add a trailing slash: `~/` shows results for me.

**Comment by valignatev** [comment]
It doesn't for me
![image](https://user-images.githubusercontent.com/7038954/123663036-ae5f8680-d83e-11eb-8913-3dabdd22d648.png)

**Comment by archseer** [comment]
Weird, is your home directory a symlink maybe?

**Comment by valignatev** [comment]
It's an absolute path

**Comment by archseer** [comment]
And `~` is the only path that has these issues?

**Comment by archseer** [comment]
The file picker follows `.gitignore`/`ignorefile` rules, maybe that's the issue?

https://github.com/helix-editor/helix/blob/51162ae6b2f57a38b5878670bf6524b24b5604e6/helix-term/src/ui/mod.rs#L76-L129

**Comment by valignatev** [comment]
Oh yeah that's definitely it, my home directory is a git repo with `*` in gitignore and forcibly added files that I track: 
https://github.com/valignatev/dotfiles/blob/master/.gitignore

I would never have guessed that "git project" is the default behavior for helix and it strictly obeys gitignore like that :)

**Comment by pickfire** [comment]
@archseer Yeah, another person like me which use `.git` for home directory.

@valignatev One suggestion I can say is used another directory for git. But yeah, that's a temporary solution for just helix. That's also good because it can hide my prompt. Now I just alias `home` so it is `~/.home` rather than `~/.git` so stuff don't seemed weird sometimes, and my prompt also does not show the branch for git when at home directory:
```fish
alias home 'git --work-tree=/home/ivan --git-dir=/home/ivan/.home'
```

**Comment by valignatev** [comment]
haha, no thanks, there is no benefit for me to introduce this complexity just to make some text editor work :)

**Comment by pickfire** [comment]
> there are lots of cases when people want to comfortably browse their gitignored files

I almost never do this unless I just press "goto declaration" or sort. We could maybe add a flag in file picker to allow browsing through hidden files.

**Comment by archseer** [comment]
Specifically in the case of the home dir, the recommendation is to use something like rcm (https://github.com/thoughtbot/rcm) or homesick. You'll definitely have issues with more than just Helix otherwise. https://askubuntu.com/questions/1316229/is-it-bad-practice-to-git-init-in-the-home-directory-to-keep-track-of-dot-files

But yes, Helix should have an option to run a filepicker without the ignore flags.

**Comment by valignatev** [comment]
I'm using this approach to manage my home directory for many years already and this is the first time something ever conflicted with it. I've used rcm and bare git repositories in the past, but all other solutions had some problems I didn't want to deal with, so I switched to this current setup that proved to be the best over the years. I'm definitely not the only one who does the same thing, here's another example: https://drewdevault.com/2019/12/30/dotfiles.html that I can recall off the top of my head.

Regardless, I think that discussions about best practices in managing dotfiles might be slightly offtopic for the issue since it seems to be related not only to the home directory, but generally to the topic of browsing ignored files.
